src/helpers/order-process/order-process-helper.ts

```diff
@@ -73,8 +84,48 @@ export const removeOrderProcesses = (
     )
   );
 
-export const updateOrderProcess = (
+export const updateOrderProcess = async (
   id: string,
-  data: Partial<OrderProcess>
-): Promise<OrderProcess> =>
-  getOrderProcessApi().updateOrderProcess(id, data) as Promise<OrderProcess>;
+  {
+    before_portfolio_item_id,
+    after_portfolio_item_id,
+    ...data
+  }: Partial<OrderProcess>
+): Promise<OrderProcess> => {
+  const op = await getOrderProcessApi().updateOrderProcess(id, {
+    name: data.name,
+    description: data.description,
+    before_portfolio_item_id: before_portfolio_item_id || '',
+    after_portfolio_item_id: after_portfolio_item_id || ''
+  });
+
+  return op as OrderProcess;
+};
+
+export const addOrderProcess = async ({
+  before_portfolio_item_id,
+  after_portfolio_item_id,
+  ...data
+}: Partial<OrderProcess>) => {
+  const op = await getOrderProcessApi().createOrderProcess({
+    name: data.name,
+    description: data.description
+  });
+
+  const promiseB =
+    before_portfolio_item_id !== undefined
+      ? getOrderProcessApi().addOrderProcessBeforeItem(
+          ((op as unknown) as OrderProcess).id as string,
+          { portfolio_item_id: before_portfolio_item_id }
+        )
+      : {};
+  const promiseA =
+    after_portfolio_item_id !== undefined
+      ? getOrderProcessApi().addOrderProcessAfterItem(
+          ((op as unknown) as OrderProcess).id as string,
+          { portfolio_item_id: after_portfolio_item_id }
+        )
+      : {};
+  await Promise.all([promiseA, promiseB]);
+  return op;
```

Comment:
I think this should return the `Promise.all([promiseA, promiseB])` promise not the order process itself. The order process entity will be outdated. It will miss the processes ids and the updated_at flag will also be stale.